packages/playwright/src/reporter.ts

```diff
@@ -177,14 +177,24 @@ export default class ReplayPlaywrightReporter implements Reporter {
 
   onBegin({ version, projects }: FullConfig) {
     const replayBrowserPath = getRuntimePath();
-    this._foundReplayBrowser = !!projects.find(
-      p => p.use.launchOptions?.executablePath === replayBrowserPath
-    );
+    for (const project of projects) {
+      this._projects[project.name] = {
+        executed: false,
+        usingReplay: project.use.launchOptions?.executablePath === replayBrowserPath,
+      };
+    }
     this.reporter.setTestRunnerVersion(version);
     this.reporter.onTestSuiteBegin();
   }
 
   onTestBegin(test: TestCase, testResult: TestResult) {
+    const projectName = test.parent.project()?.name;
+
+    // it's important to handle the root project's name here and that's an empty string
+    if (typeof projectName === "string") {
```

Comment:
This comment and the check below are not totally aligned. An empty string would still be `typeof` "string"

Do you mean to check `if (projectName)` ?

Reply:
no, the opposite - I meant that the root project's name is an empty string and thus I have to check `typeof` instead of relying on coercion

Reply:
Ah, okay. I misinterpreted the intention of the comment.